### ISSUE TITLE
Running several times gives different results

### ISSUE THREAD

**Issue by rybchuk** [path1380/Homework3_Grp_2#3]
Running `lengendre.p` multiple times in a row appends errors to `output.txt`. Because we are testing 3 functions right now, 3 lines should always be appended. This is not always the case.

**Comment by rybchuk** [comment]
Example:
My last call only appended two lines. I got the following message in between the first and second compile.

```
./main.x >> output.txt
Operating system error: Cannot allocate memory
Allocation would exceed memory limit
Makefile_main:11: recipe for target 'main.x' failed
make: *** [main.x] Error 1
```

**Comment by rybchuk** [comment]
Full error message:

```
alex@laserlab:~/Classwork/APPM5720/Homework3_Grp_2$ perl legendre.p 
gfortran -fbounds-check -Wall -fbacktrace -g -c type_defs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c quad_1dmod.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c leg_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c approx_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c InputControl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c lgl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c coeff.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c main.f90
gfortran -o main.x type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o
./main.x >> output.txt
Operating system error: Cannot allocate memory
Allocation would exceed memory limit
Makefile_main:11: recipe for target 'main.x' failed
make: *** [main.x] Error 1
gfortran -fbounds-check -Wall -fbacktrace -g -c InputControl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c main.f90
gfortran -o main.x type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o
./main.x >> output.txt
 0.3108624468950438E-14 , 0.4925187593944632E-14 
rm -f type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o main.x *.mod
gfortran -fbounds-check -Wall -fbacktrace -g -c type_defs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c quad_1dmod.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c leg_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c approx_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c InputControl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c lgl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c coeff.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c main.f90
gfortran -o main.x type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o
./main.x >> output.txt
 0.3108624468950438E-14 , 0.4925187593944632E-14 
rm -f type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o main.x *.mod
```

Successful run:
```
alex@laserlab:~/Classwork/APPM5720/Homework3_Grp_2$ perl legendre.p 
gfortran -fbounds-check -Wall -fbacktrace -g -c type_defs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c quad_1dmod.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c leg_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c approx_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c InputControl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c lgl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c coeff.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c main.f90
gfortran -o main.x type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o
./main.x >> output.txt
 0.3108624468950438E-14 , 0.4925187593944632E-14 
rm -f type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o main.x *.mod
gfortran -fbounds-check -Wall -fbacktrace -g -c type_defs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c quad_1dmod.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c leg_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c approx_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c InputControl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c lgl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c coeff.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c main.f90
gfortran -o main.x type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o
./main.x >> output.txt
 0.3108624468950438E-14 , 0.4925187593944632E-14 
rm -f type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o main.x *.mod
gfortran -fbounds-check -Wall -fbacktrace -g -c type_defs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c quad_1dmod.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c leg_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c approx_funs.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c InputControl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c lgl.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c coeff.f90
gfortran -fbounds-check -Wall -fbacktrace -g -c main.f90
gfortran -o main.x type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o
./main.x >> output.txt
 0.3108624468950438E-14 , 0.4925187593944632E-14 
rm -f type_defs.o quad_1dmod.o leg_funs.o approx_funs.o InputControl.o lgl.o coeff.o main.o main.x *.mod
```

**Comment by fortinog** [comment]
That's odd, are you running it on a virtual machine by any chance? Do you happen to know how many gigs of RAM your environment has? I had issues with multiple runs on a separate virtualmachine I was using to test memory issues but the issues went away when I started compiling on my mac with its full RAM.

**Comment by rybchuk** [comment]
I'm running it on a full OS, so no virtual machine. I found one stackoverflow post that suggests this might be a Linux specific issue.

**Comment by fortinog** [comment]
Quick update: I was testing this out in my Ubuntu VM and it turns out that adding the statement interval_info(i)%nvars = 1
allowed me to run the perl script multiple times without any memory issues. I guess this is just a Linux specific issue and the built Intel gfortran compiler works around it.

**Comment by rybchuk** [comment]
Sadly that fix didn't work for me.

It looks like there was a big change going from F90 -> F2003 in the way that arrays are handled in user defined types, [link](https://kiwi.atmos.colostate.edu/fortran/docs/fortran2012.key-8.pdf) and [link](http://www.cs.rpi.edu/~szymansk/OOF90/bugs.html#5) and [best link](http://people.ds.cam.ac.uk/nmm1/fortran/paper_09.pdf). I'm digging into this.

Also I think we're using the same compiler. Intel's fortran compiler (I believe) is called ifort, and gfortran was developed by the gcc people.

EDIT:
This issue was resolved in b31ef33c0c136d6e0f993e75b2e937af2e180449.
